src/jsonvalidator.ts

```diff
@@ -1,89 +1,111 @@
-        .prop('baseDomain', S.string().required())
-      ).required()
-      .prop('Cicd', S.object()
-        .prop('env', S.object()
-          .prop('account', S.string().required())
-          .prop('region', S.string().required())
-        ).required()
-        .prop('profile', S.string().required())
-        .prop('accounts', S.array().required())
-        .prop('branchName', S.string().required())
-        .prop('gitHubTokenSecretArn', S.string().required())
-        .prop('codeCommitUserPublicKey', S.string().required())
-        .prop('repositoryProvider', S.string().required())
-        .prop('baseDomain', S.string().required())
-      ).required()
-      .prop('FederationAccounts', S.array().minItems(1).required())
-      .prop('lastGenerated', S.string())
-      .prop('DomainManager', S.object()
-        .prop('env', S.object()
-          .prop('account', S.string().required())
-          .prop('region', S.string().required())
-        ).required()
-        .prop('profile', S.string().required())
-        .prop('baseDomain', S.string().required())
-        .prop('hostedZoneId', S.string().required())
-      ).required()
-      .prop('Staging', S.object()
-        .prop('env', S.object()
-          .prop('account', S.string().required())
-          .prop('region', S.string().required())
-        ).required()
-        .prop('profile', S.string().required())
-        .prop('withDomain', S.boolean().required())
-        .prop('webAppUrl', S.string().required())
-        .prop('requireManualApproval', S.boolean().required())
-      ).required()
-      .prop('Production', S.object()
-        .prop('env', S.object()
-          .prop('account', S.string().required())
-          .prop('region', S.string().required())
-        ).required()
-        .prop('profile', S.string().required())
-        .prop('withDomain', S.boolean().required())
-        .prop('webAppUrl', S.string().required())
-        .prop('requireManualApproval', S.boolean().required())
-      ).required()
-    ).required()
+import * as path from 'path'
+/* This function programmatically defines the default.json file */
+export function getDefaultConfigSchema (): object {
+  return S.object()
+    .id('default-config')
+    .title('Default Config File Schema')
+    .description('Default Config File Schema')
+    .prop(
+      'app',
+      S.object()
+        .prop('prefix', S.string().required())
+        .prop('name', S.string().required())
+    )
+    .required()
+    .prop(
+      'cicd',
+      S.object()
+        .prop('target', S.string())
+        .prop('buildspecFile', S.string())
+        .prop('permissionsFile', S.string())
+        .prop('provider', S.string())
+        .prop('profile', S.string())
+        .prop('repositoryUrl', S.string())
+        .prop('branchName', S.string())
+        .prop('codeCommitUserPublicKey', S.string())
+        .prop('environmentVariables', S.array())
+        .prop('stages', S.array())
+    )
+    .prop(
+      'accounts',
```

Comment:
All account keys like `cicd`, `deafault`, or `staging` are arbitrary and depend on the user's decision.
The rule is that if there's a 'target' defined in dev.target or cicd.target or cicd.stages[].target, given value name have to be listed as a property in the accounts section. 

As this is a rather dynamic relation I'm not sure if that can be codified with a schema. 
What would you say to keep schema for required items in the config, and the rest to somehow check with custom code?